contracts/test/utils.ts

```diff
@@ -117,11 +119,7 @@ const buildTestFn = (fnNameOrObject: FnNameOrObject, overrides = {}, debug = fal
         console.log('')
       }
 
-      if (fn.shouldFail) {
-        assert(failed, `${fn.method} should have failed`)
-      } else {
-        assert(!failed, `${fn.method} failed`)
-      }
```

Comment:
I think if any of those ds-test assertions fail, the test should fail. Assertions should be placed such that they will not fail.